components/brave_shields/core/browser/ad_block_component_filters_provider.cc

```diff
@@ -97,9 +99,13 @@ void AdBlockComponentFiltersProvider::OnComponentReady(
   NotifyObservers(engine_is_default_);
 
   if (!old_path.empty()) {
-    base::ThreadPool::PostTask(
-        FROM_HERE, {base::TaskPriority::BEST_EFFORT, base::MayBlock()},
-        base::BindOnce(IgnoreResult(&base::DeletePathRecursively), old_path));
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
```

Comment:
I'm a bit confused here, shouldn't `path` (and therefore `old_path`) be correct whether it's a test or not?

Reply:
Without the special case, the relevant files under `test/data/` get deleted from disk during tests because of this cleanup logic. Then the git repo ends up with a dirty state and subsequent tests fail.

I do think it's a little bit weird; if you have a better solution please let me know :sweat_smile:

Reply:
I guess the other option is to recursively copy all the relevant files to a new temp dir at the beginning of each test. Also a little weird considering it's otherwise read-only data, but it could simplify the non-test logic?